### PR TITLE
Fix GDAL and bump version of DGGRID7

### PR DESCRIPTION
**Pull request by danlooo**
This PR aims to fix a version mismatch of DGGRID7_jll and GDAL_jll by updating DGGRID7_jll to the latest official release 7.7 of [DGGRID](https://github.com/sahrk/DGGRID/) that works with BinaryBuilder. @allixender what do you think?

## Error

Executing in docker image julia:1.8.5-bullseye :

```{julia}
using Pkg
Pkg.add("DGGRID7_jll")

using DGGRID7_jll

download("https://raw.githubusercontent.com/sahrk/DGGRID/4f24e9de46f9e669af4cba94d9d148b557a1de11/examples/icosahedron/icosahedron.meta", "meta.txt")
mkdir("outputfiles")

DGGRID7_jll.dggrid() do dggrid_path
    run(`$dggrid_path meta.txt`)
end
```

results in error

```
/root/.julia/artifacts/3381b9f883af7e984906191545c4fa939e07e91b/bin/dggrid: error while loading shared libraries: libgdal.so.28: cannot open shared object file: No such file or directory
```

Indeed, dggrid requires libgdal.so.28, but package GDAL_jll provides libgdal.so.32.3.6.2:

```
root@6e8b0f893490:~/.julia/artifacts# find ~/.julia/artifacts/6f97fbb4bedf23ccd1d7757586cb7659222a0465/lib
/root/.julia/artifacts/6f97fbb4bedf23ccd1d7757586cb7659222a0465/lib
/root/.julia/artifacts/6f97fbb4bedf23ccd1d7757586cb7659222a0465/lib/pkgconfig
/root/.julia/artifacts/6f97fbb4bedf23ccd1d7757586cb7659222a0465/lib/pkgconfig/gdal.pc
/root/.julia/artifacts/6f97fbb4bedf23ccd1d7757586cb7659222a0465/lib/libgdal.so.32
/root/.julia/artifacts/6f97fbb4bedf23ccd1d7757586cb7659222a0465/lib/gdalplugins
/root/.julia/artifacts/6f97fbb4bedf23ccd1d7757586cb7659222a0465/lib/gdalplugins/drivers.ini
/root/.julia/artifacts/6f97fbb4bedf23ccd1d7757586cb7659222a0465/lib/libgdal.so
/root/.julia/artifacts/6f97fbb4bedf23ccd1d7757586cb7659222a0465/lib/cmake
/root/.julia/artifacts/6f97fbb4bedf23ccd1d7757586cb7659222a0465/lib/cmake/gdal
/root/.julia/artifacts/6f97fbb4bedf23ccd1d7757586cb7659222a0465/lib/cmake/gdal/GDALConfig.cmake
/root/.julia/artifacts/6f97fbb4bedf23ccd1d7757586cb7659222a0465/lib/cmake/gdal/GDALConfigVersion.cmake
/root/.julia/artifacts/6f97fbb4bedf23ccd1d7757586cb7659222a0465/lib/cmake/gdal/GDAL-targets.cmake
/root/.julia/artifacts/6f97fbb4bedf23ccd1d7757586cb7659222a0465/lib/cmake/gdal/GDAL-targets-release.cmake
/root/.julia/artifacts/6f97fbb4bedf23ccd1d7757586cb7659222a0465/lib/libgdal.so.32.3.6.2
```

## Fix

This PR uses the latest official release 7.7 of [DGGRID](https://github.com/sahrk/DGGRID/) that works with BinaryBuilder.

Version 7.8 of DGGRID results in error `/workspace/srcdir/DGGRID/src/lib/dglib/lib/DgZOrderRF.cpp:96:24: error: expected ‘)’ before ‘PRIx64‘`

## Test

julia build_tarballs.jl successed to run on a local Linux machine.
Furthermore, products/DGGRID7.v0.9.0.x86_64-linux-gnu-cxx11.tar.gz was tested by integration test [doexamples.sh](https://github.com/sahrk/DGGRID/blob/4f24e9de46f9e669af4cba94d9d148b557a1de11/examples/doexamples.sh). Exit code was 0, but output files still contain these errors:

```
$ find ~/repos/Yggdrasil/D/DGGRID7/DGGRID/examples -type f | grep outputfiles | xargs -i grep -H ERROR {}

Yggdrasil/D/DGGRID7/DGGRID/examples/planetRiskGridGen/outputfiles/planetRiskGridGen.txt:FATAL ERROR: Invalid parameter data in parameter:
Yggdrasil/D/DGGRID7/DGGRID/examples/planetRiskClipHiRes/outputfiles/planetRiskClipHiRes.txt:FATAL ERROR: Invalid parameter data in parameter:
Yggdrasil/D/DGGRID7/DGGRID/examples/gdalCollection/outputfiles/gdalCollection.txt:FATAL ERROR: DgParamList::setParam() unknown parameter collection_output_gdal_format
Yggdrasil/D/DGGRID7/DGGRID/examples/gdalExample/outputfiles/gdalExample.txt:FATAL ERROR: Invalid parameter data in parameter:
Yggdrasil/D/DGGRID7/DGGRID/examples/holes/outputfiles/holes.txt:FATAL ERROR: Invalid parameter data in parameter:
Yggdrasil/D/DGGRID7/DGGRID/examples/planetRiskClipMulti/outputfiles/planetRiskClipMulti.txt:FATAL ERROR: Invalid parameter data in parameter:
Yggdrasil/D/DGGRID7/DGGRID/examples/planetRiskGridNoWrap/outputfiles/planetRiskGridNoWrap.txt:FATAL ERROR: Invalid parameter data in parameter:
```

### CODE REVIEW

File: D/DGGRID7/build_tarballs.jl [hunk 1]
```diff
@@ -7,25 +7,29 @@ version = v"0.9.0"
...
+    GitSource("https://github.com/sahrk/DGGRID.git", "4f24e9de46f9e669af4cba94d9d148b557a1de11")
 ]
 
 # Bash recipe for building across all platforms
 script = raw"""
-cd $WORKSPACE/srcdir/DGGRID/src/
+cd $WORKSPACE/srcdir/DGGRID
+mkdir build
+cd build
+cmake -DCMAKE_BUILD_TYPE=Release ..
```
Comment (giordano) [inline D/DGGRID7/build_tarballs.jl:18]: https://docs.binarybuilder.org/stable/build_tips/#CMake-builds

File: D/DGGRID7/build_tarballs.jl [hunk 2]
```diff
@@ -7,25 +7,29 @@ version = v"0.9.0"
...
 if [[ "${target}" == x86_64-linux-musl ]]; then
     # Remove libexpat to avoid it being picked up by mistake
     rm /usr/lib/libexpat.so*
 fi
 make -j${nproc} CCOMP="${CC}" CPPCOMP="${CXX}"
-cp "apps/dggrid/dggrid${exeext}" "${bindir}/."
+mkdir -p ${bindir}
+cp "src/apps/dggrid/dggrid${exeext}" "${bindir}/."
```
Comment (giordano) [inline D/DGGRID7/build_tarballs.jl:25]:
```suggestion
install -Dvm 755 "src/apps/dggrid/dggrid${exeext}" "${bindir}/dggrid${exeext}"
```

File: D/DGGRID7/build_tarballs.jl [hunk 3]
```diff
@@ -37,4 +41,4 @@ dependencies = [
 ]
 
 # Build the tarballs, and possibly a `build.jl` as well.
-build_tarballs(ARGS, name, version, sources, script, platforms, products, dependencies; preferred_gcc_version = v"8.1.0")
+build_tarballs(ARGS, name, version, sources, script, platforms, products, dependencies; preferred_gcc_version=v"8.1.0", julia_compat="v1.6.0")
```
Comment (giordano) [inline D/DGGRID7/build_tarballs.jl:44]: To change the `julia_compat` you ***must*** change the version number: #2763

**Comment by allixender** [comment]
Hi, thanks for the preps, much appreciated. Updating DGGRID would be great. They changed the build to cmake, and a GDAL bump to at least 3.3+ would also be great. The errors in the examples might point to linking to an outdated GDAL version. How can I help with something?

**Comment by danlooo** [comment]
Thanks for the comments! I just set GDAL version to v301.600.20, because file `libgdal.so.32` is present in [the package](https://github.com/JuliaBinaryWrappers/GDAL_jll.jl/releases/download/GDAL-v301.600.200%2B0/GDAL.v301.600.200.x86_64-linux-gnu-cxx11.tar.gz)

File: D/DGGRID7/build_tarballs.jl (hunk 3 above)
Comment (danlooo) [reply D/DGGRID7/build_tarballs.jl:43]: `julia_compat="v1.6.0"` fixed in 9c5c12de1f46e09bfb63c41bd04847164d842e7f

Comment (giordano) [reply D/DGGRID7/build_tarballs.jl:43]: The version number of the packag, as explained in the non-negotiable requirements in issue #2763 linked above

Comment (danlooo) [reply D/DGGRID7/build_tarballs.jl:43]: ok. See f9bf39bcf252f89420ec258038f477e0f143c2d5

File: D/DGGRID7/build_tarballs.jl [hunk 4]
```diff
@@ -7,34 +7,37 @@ version = v"0.9.0"
...
+mkdir build
+cd build
+cmake -DCMAKE_INSTALL_PREFIX=${prefix} -DCMAKE_TOOLCHAIN_FILE=${CMAKE_TARGET_TOOLCHAIN} -DCMAKE_BUILD_TYPE=Release ..
 if [[ "${target}" == x86_64-linux-musl ]]; then
     # Remove libexpat to avoid it being picked up by mistake
     rm /usr/lib/libexpat.so*
 fi
 make -j${nproc} CCOMP="${CC}" CPPCOMP="${CXX}"
-cp "apps/dggrid/dggrid${exeext}" "${bindir}/."
+install -Dvm 755 "src/apps/dggrid/dggrid${exeext}" "${bindir}/dggrid${exeext}"
```
[the remark this answers is not in the document]
Comment (allixender) [reply D/DGGRID7/build_tarballs.jl:24]: I think there is no make install.

[the remark this answers is not in the document]
Comment (danlooo) [reply D/DGGRID7/build_tarballs.jl:24]: Yes:
```
/usr/bin/cmake -E cmake_progress_start /workspace/srcdir/DGGRID/build/CMakeFiles 0
make: *** No rule to make target 'install'.  Stop.
```

**Comment by allixender** [comment]
> Unfortunately, DGGRID v7.8 does not compile:
> 
> ```
> /workspace/srcdir/DGGRID/src/lib/dglib/lib/DgZOrderRF.cpp:96:24: error: expected ‘)’ before ‘PRIx64’
>     if (!sscanf(tok, "%" PRIx64, &val))
>                ~        ^~~~~~~
> ```
> 
> See [dggrid_v7.8.log](https://github.com/JuliaPackaging/Yggdrasil/files/11463999/dggrid_v7.8.log) and [this branch](https://github.com/danlooo/Yggdrasil/blob/dggrid_v7.8/D/DGGRID7/build_tarballs.jl) for details. I encountered similar erros while working on julia bindings in project [ddgrid-julia](https://github.com/danlooo/dggrid-julia). However, resolving this looks like a new issue.

Whoa, you are already on the way, amazing. I'll see if I can reproduce this and I will send something to Kevin.

**Pull request merged**